### PR TITLE
Derive more defaults in chain config

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

- Attempt to detect RPC URL, chain name, and chain ID

### Testing

```sh
Creating a new chain config
? Detected rpc url as http://localhost:8545, is this correct? no
? Enter http or https rpc url http://localhost:8546
? Detected chain name as anvil8546, is this correct? yes
? Detected chain id as 31338, is this correct? yes
? Do you want to set block or gas properties for this chain config?(optional) (y/N)
```

### CODE REVIEW

File: typescript/cli/src/config/chain.ts [hunk 1]
```diff
@@ -38,19 +39,27 @@ export async function createChainConfig({
...
+      message: 'Enter chain name (one word, lower case)',
+    });
+  }
+
   const metadata: ChainMetadata = {
     name,
-    chainId: parseInt(chainId, 10),
-    domainId: parseInt(domainId, 10),
+    chainId,
+    domainId: chainId,
```
Comment (nbayindirli) [inline typescript/cli/src/config/chain.ts:58]: ❤️

File: typescript/cli/src/config/chain.ts [hunk 2]
```diff
@@ -38,19 +39,27 @@ export async function createChainConfig({
...
+
   const rpcUrl = await input({ message: 'Enter http or https rpc url' });
+  const network = await new ethers.providers.JsonRpcProvider(
+    rpcUrl,
+  ).getNetwork();
+
+  let { chainId, name } = network;
+  if (name === 'unknown') {
+    name = await input({
+      message: 'Enter chain name (one word, lower case)',
```
Comment (nbayindirli) [inline typescript/cli/src/config/chain.ts:51]: possible nit: does input auto-add a colon at the end here?

**Review by nbayindirli — APPROVED** [review]
lgtm

File: typescript/cli/src/config/chain.ts (hunk 2 above)
Comment (yorhodes) [reply typescript/cli/src/config/chain.ts:51]: idk but would prefer not to change this everywhere

Comment (jmrossy) [reply typescript/cli/src/config/chain.ts:51]: Should we really rely on the name provided by the network given that we've invented our own naming type:
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/typescript/sdk/src/metadata/customZodTypes.ts#L20

**Pull request merged**